python/ql/src/semmle/python/ApiGraphs.qll

```diff
@@ -445,6 +440,51 @@ module API {
       m = n.getEnclosingModule()
     }
 
+    /**
+     * Holds if `n` is an access of a variable called `name` (which is _not_ the name of a
+     * built-in, and which is _not_ a global defined in the enclosing module) inside the scope `s`.
+     */
+    private predicate name_possibly_defined_in_import_star(NameNode n, string name, Scope s) {
+      n.isLoad() and
+      name = n.getId() and
+      // Not already defined in an enclosing scope.
+      not exists(LocalVariable v |
+        v.getId() = name and v.getScope().getEnclosingScope*() = n.getScope()
```

Comment:
Shouldn't the `.getEnclosingScope*()` bit be on the other side?

With the current code, wouldn't we reject the `foo()` call as being `mymodule.foo()`, since you can reach it's scope by going through the enclosing scopes from `foo = "FOO"`?

```py
from mymodule import *

def func1():
    foo()
    def func2():
        foo = "FOO"
```

So that it should be this instead

```suggestion
        v.getId() = name and v.getScope() = n.getScope().getEnclosingScope*()
```

Thereby not modeling the `foo()` call to be `mymodule.foo()` in the following code

```py
from mymodule import *

def func1():
    foo = print
    def func2():
        foo()
```

and maybe we can add such a test-case as well? :blush: